docs/proposals/self-hosted-kubelet.md

```diff
@@ -0,0 +1,75 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Proposal: Self-hosted kubelet
+
+## Abstract
+
+In a self-hosted Kubernetes deployment, we have the initial bootstrap problem. This proposal presents a solution to the kubelet bootstrap, and assumes a functioning control plane, and a kubelet that can securely contact the API server.
```

Comment:
I guess I was thinking about this from the perspective of "in terms of this proposal, assume a functioning apiserver" as a means of keeping the discussion more focused. Ultimately the apiserver could be a static pod, or just a binary run directly on the host, or docker container outside k8s, etc.

But you're right that you wouldn't strictly need an apiserver to demonstrate the same functionality. The kubelet pod could be a static pod as well (would be an odd use-case, but should work assuming the kubelet pod doesn't need secrets/configMap/etc). So maybe just drop that line, as it is somewhat orthogonal? 

Also, agree that it would be helpful to add a "motivations" section to cover reasons we want self-hosted kubelet.